src/legacy/core_plugins/console/server/live_hosts_manager/utils.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import fetch from 'node-fetch';
+
+const probe = async (host: string): Promise<{ ok: boolean }> => {
+  try {
+    await fetch(`${host}/`, { method: 'HEAD' });
+    return { ok: true };
+  } catch (e) {
+    return { ok: false };
+  }
+};
+
+const roundRobinNextIdx = (number: number, max: number) => {
+  if (number >= max) {
+    return 0;
+  }
+  return number + 1;
+};
+
+export const findLiveHostIdx = async (startIdx: number, hosts: string[]) => {
+  if ((await probe(hosts[startIdx])).ok) {
```

Comment:
any thoughts on if its possible to select the next live host if a request fails?  not a huge issue, but the preflight probe approach will 2x latency

Reply:
That's a really good point! This is based on how I _think_ the current ES Node client works, but it may be sub-optimal to do a manual pre-flight for every request.

Additionally, the current implementation has an internal pointer to the last live host which it will use again until that host fails and it tries to find a new one. This makes it slightly indeterminate regarding which host we are talking to. One alternative could be:

1. We optimistically try a host (no preflight `HEAD /` request)
2. We scope cycling through hosts to a specific request. This will remove the internal pointer and means that we cycle through the hosts first to last for every request.
3. Failure case is when all hosts have responded with `ECONNREFUSED` (or something like that? - need to double check) we don't want to swallow legitimate error responses

What do you think?

Reply:
that sounds great, makes sense to me :)